company-coq.el

```diff
@@ -2201,11 +2192,19 @@ in that file."
   (concat "\\`" (regexp-opt headers) "\\_>[\n[:space:]]+" "\\(" company-coq-symbol-regexp "\\)"))
 
 (defun company-coq--loc-fully-qualified-name (fqn)
-  "Find source file for fully qualified name FQN."
-  (let* ((spec (company-coq-longest-matching-path-spec fqn))
-         (logical (if spec (concat (car spec) ".") ""))
-         (mod-name (replace-regexp-in-string "\\..*\\'" "" fqn nil nil nil (length logical))))
-    (company-coq-library-path logical mod-name spec)))
+  (cl-labels
+      ((recs (fqn)
+	     (message "fqn-loc %s" fqn)
+	     (let* ((output (company-coq-ask-prover-swallow-errors (format company-coq-locate-lib-cmd fqn))))
+	       (or (and output
+			(save-match-data
+			  (when (and (string-match company-coq-locate-lib-output-format output)
+				     (string-match-p company-coq-compiled-regexp (match-string-no-properties 3 output)))
+			    (cons (match-string-no-properties 2 output) fqn))))
+		   (let* ((prefixfqn (file-name-sans-extension fqn)))
+		     (if (< (length prefixfqn) (length fqn)) (recs prefixfqn)))))))
+    (recs fqn)));; do if file exists as it was done before
+;; (replace-regexp-in-string "_build/default" "" (recs fqn) nil 'literal)
```

Comment:
to be on the safer side, check that the parent directory of `_build` has a `dune-project` file